referenceframe/frame_system.go

```diff
@@ -397,6 +401,44 @@ func (sfs *simpleFrameSystem) getFrameToWorldTransform(inputMap map[string][]Inp
 	return srcToWorld, err
 }
 
+// ReplaceFrame transfers replaceMe's children and parentage to replaceWith. ReplaceMe is removed entirely from the frame system.
+// ReplaceWith is not allowed to exist within the frame system at the time of the call.
+func (sfs *simpleFrameSystem) ReplaceFrame(fs FrameSystem, replaceMe, replaceWith Frame) error {
+	if len(fs.FrameNames()) == 0 {
+		return errors.New("cannot replace frames within an empty framesystem")
+	}
+	if replaceMe.Name() == replaceWith.Name() {
+		return errors.New("replaceWith cannot have the same name as replaceMe")
+	}
+	if fs.Frame(replaceWith.Name()) != nil {
```

Comment:
I think these three checks can be replaced by two:  "replaceMe must be in the frame system" and "replaceWith must not be in the frame system"

Reply:
And a third, World cannot be replaced

Reply:
we will need to enforce that `replaceMe` and `replaceWith` have different names with:
```
	if replaceMe.Name() == replaceWith.Name() {
		return fmt.Errorf("%s (replaceMe) cannot have the same name as %s (replaceWith)", replaceWith.Name(), replaceMe.Name())
	}
```
Otherwise we do not know which of `replaceMe`, `replaceWith` already exist within the framesystem

Reply:
I don't think that needs to be enforced; on the contrary, I think we would want to encourage them having the same name. Otherwise things like collision specifications will not work well.

So:
- World cannot be replaced.
- replaceMe must be in the frame system
- replaceWith must not be in the frame system, unless it has the same name as replaceMe

Reply:
Actually, thinking about this more, I think we may want to require that `replaceWith` must have an identical name to `replaceMe`.

Reply:
Yeah I agree I think it should probably have the same name, that makes it so less is invalidated through these changes (still things like input of the fs will be invalidated if DOF changes but at least this gets partway there)